tests/plugins/test_accounts.py

```diff
@@ -99,6 +101,48 @@ def test_cannot_create_other_account_if_authenticated(self):
                                   status=400)
         assert 'do not match' in resp.json['message']
 
+    def test_authentication_does_not_call_bcrypt_twice(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        with mock.patch('kinto.plugins.accounts.authentication.bcrypt') as mocked_bcrypt:
+            resp = self.app.get('/', headers=get_user_headers('me', 'bouh'))
+            assert resp.json['user']['id'] == 'account:me'
+
+            resp = self.app.get('/', headers=get_user_headers('me', 'bouh'))
+            assert resp.json['user']['id'] == 'account:me'
+
+            mocked_bcrypt.checkpw.assert_called_once()
+
+    def test_authentication_check_bcrypt_again_if_password_changes(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        with mock.patch('kinto.plugins.accounts.authentication.bcrypt') as mocked_bcrypt:
+            resp = self.app.get('/', headers=get_user_headers('me', 'bouh'))
+            assert resp.json['user']['id'] == 'account:me'
+
+            self.app.patch_json('/accounts/me', {'data': {'password': 'blah'}},
+                                status=200, headers=get_user_headers('me', 'bouh'))
+
+            resp = self.app.get('/', headers=get_user_headers('me', 'blah'))
+            assert resp.json['user']['id'] == 'account:me'
+
+            assert mocked_bcrypt.checkpw.call_count == 2
+
+    def test_authentication_refresh_the_cache_each_time_we_authenticate(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        resp = self.app.get('/', headers=get_user_headers('me', 'bouh'))
+        assert resp.json['user']['id'] == 'account:me'
+
+        time.sleep(2)
```

Comment:
We should not slow the tests by two seconds here, we can do better like call `.cache.expire(cache_key, 1)` first and then make sure it's `>1` or whatever

Reply:
I don't understand how setting cache.expire helps here can you elaborate?

Reply:
Ok got it